### PR TITLE
workflow: set unmerged inspiral and psd files as non critical

### PR DESCRIPTION
**Pull request by tdent**
The merged files contain all info necessary to proceed with the rest of the pipeline, so dont store the individula inspiral files by default.  This should make NFS happier not to have to deal with 100000's of files ad infinitum.

### CODE REVIEW

**Comment by spxiwh** [comment]
Hi Tom,

If you do this though, you basically remove most of the workflow's reuse functionality. Ie. if you need to rerun one inspiral job (because DQ change or ....), you now need to rerun *all* inspiral jobs.

**Comment by tdent** [comment]
No, you remove the reuse functionality *if* inspiral jobs need to be rerun, which is only the case when 'CAT1' changes (ie in analyzed segments) are being considered. 
So changes in DQ vetoes or coinc options or ranking statistics or postproc or plotting are still fine. 
I claim it *is* useful to be able to do 'CAT234' type reruns with much more efficient file use.  (Eg it has been useful to me investigating binning for IMBH.) 
So, there should be a file-keeping-setting for which all inspiral files are kept, and that should be the default for runs over recent data, but there should also be a file-keeping-setting where only merged files are kept, eg for reruns over data with stable CAT1 segments. 
At the moment you don't have the choice to do the latter.

**Comment by spxiwh** [comment]
Hi Tom,

Okay, it makes sense to have level 2 be to keep INSPIRAL jobs, and level 3 keeping only the merged products. However, I would like the default to be that we keep all INSPIRAL jobs. Currently the ini files say:

file-retention-level = all_triggers

which would not keep INSPIRAL triggers by default. 

*OR* make the MERGED files level 4 (results_only).

Cheers
Ian

**Comment by tdent** [comment]
Yes, so we need suitable names for the keep-merged-trigger-files choice and the keep-all-inspirals choice and make the second one the de facto choice in the standard ini files but indicate that other choices will work if the CAT1 segments do not change.

**Comment by tdent** [comment]
Hi Ian, I've added a bunch of further commits to change the names of the retention levels and reassign a couple of levels (mainly for intermediate coinc output files).  'all_triggers' would remain the default for initial runs, but 'merged_triggers' would be appropriate for runs on stable data segments where only 'CAT2' and/or coinc-level changes are expected.  
Any feedback?

File: pycbc/workflow/legacy_ihope.py [hunk 1]
```diff
@@ -508,7 +508,7 @@ class LegacyCohPTFSbvPlotter(LegacyAnalysisExecutable):
     The class responsible for setting up jobs for legacy coh_PTF_sbv_plotter
     executable.
     """
-    current_retention_level = Executable.CRITICAL
+    current_retention_level = Executable.FINAL_RESULT
```
Comment (spxiwh) [inline pycbc/workflow/legacy_ihope.py]: Thanks for fixing mis-labeled GRB executables!

**Comment by spxiwh** [comment]
Hi Tom,

Thanks for doing this, and apologies for the delay in looking at this! I am happy with the code changes as is, but as this does make changes in core.py we should run the test workflow (ie. run it end-to-end ... it only takes ~ hour).

Instructions for running this can be found here:

https://code.pycbc.phy.syr.edu/ligo-cbc/pycbc-config/tree/master/O1/workflow_test

Please post the result page when done. Assuming nothing goes run I will be happy to merge.

Cheers
Ian

**Comment by tdent** [comment]
Am currently running the test workflow but none of the calculate_psd jobs are running because Atlas and 4 cores.  Oh well.  BTW, when this one finishes I will try with the new retention level ('merged_triggers') to check that that does what is advertised.

**Comment by tdent** [comment]
output for 'all_triggers' run is here : https://www.atlas.aei.uni-hannover.de/~tdent/LSC/tests/pycbc_workflow/ 
'merged_triggers' run is currently underway.

**Pull request merged**